src/libbluechi/common/time-util.c

```diff
@@ -15,23 +15,15 @@ uint64_t get_time_micros() {
         if (clock_gettime(CLOCK_REALTIME, &now) < 0) {
                 return 0;
         }
-        uint64_t now_micros = now.tv_sec * sec_to_microsec_multiplier +
-                        (uint64_t) ((double) now.tv_nsec * nanosec_to_microsec_multiplier);
-        return now_micros;
+        return timespec_to_micros(now);
 }
 
-int get_time_seconds(time_t *ret_time_seconds) {
-        if (ret_time_seconds == NULL) {
-                return -EINVAL;
-        }
-
+uint64_t get_time_micros_monotonic() {
```

Comment:
I'd prefer to keep the (systemd-like) style of returning the error code and using the parameter for returning. This way, the error logs where this function is called provides more information. 
```suggestion
int get_time_micros_monotonic(time_t *ret_time_seconds) {
```

The same applies for `timespec_to_micros` - maybe using `EINVAL` as return code?!
```
int timespec_to_micros(struct timespec time, uint64_t *ret){
   ...
   if (sec_micros > UINT64_MAX - nsec_micros) {
      return -EINVAL;
   }
   ...
```

Reply:
If you perfer systemd-like style, what do you think about reusing code from time-util.c of systemd? It uses assert() for error cases and returns USEC_INFINITY(maximum value) for overflow or abnormal values.

e.g. systemd src/basic/time-util.c
```
usec_t now(clockid_t clock_id) {
        struct timespec ts;

        assert_se(clock_gettime(map_clock_id(clock_id), &ts) == 0);

        return timespec_load(&ts);
}
...
usec_t timespec_load(const struct timespec *ts) {
        assert(ts);

        if (ts->tv_sec < 0 || ts->tv_nsec < 0)
                return USEC_INFINITY;

        if ((usec_t) ts->tv_sec > (UINT64_MAX - (ts->tv_nsec / NSEC_PER_USEC)) / USEC_PER_SEC)
                return USEC_INFINITY;

        return
                (usec_t) ts->tv_sec * USEC_PER_SEC +
                (usec_t) ts->tv_nsec / NSEC_PER_USEC;
}
```

Reply:
Makes sense to me. 
I probably wouldn't import `assert_se` from systemd, but use a plain `assert` when `clock_gettime` fails - that operation should never fail. Returning `USEC_INFINITY` in overflow/abnormal value cases is a good idea, I think. Could you define `USEC_INFINITY` in `time-util.h`? We have constants like `USEC_PER_SEC` already in [protocol.h](https://github.com/eclipse-bluechi/bluechi/blob/main/src/libbluechi/common/protocol.h#L10-L13), but to me that seems off now. I'd move these to `time-util.h` in a later PR.

Reply:
Updated.

Reply:
Thanks, looks good!